src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -202,28 +200,23 @@ class CosmasGoogleCloudService(private val bucketName: String,
             responseObserver.onError(StatusException(requestStatus))
             return
         }
-        while (true) {
-            askedVersion++
-            val blob: Blob = try {
-                this.storage.get(BlobInfo.newBuilder(this.bucketName, request.getVersionRequest.fileId).build().blobId,
-                        Storage.BlobGetOption.generationMatch(askedVersion))
-            } catch (e: StorageException) {
-                handleStorageException(e, responseObserver)
-                return
-            } ?: break
-            val fileVersion = CosmasProto.FileVersion.parseFrom(blob.getContent())
-            patchList.addAll(fileVersion.patchesList)
-        }
+        val textNoPatch : String
+        try {
         val textBeforeCandidateDelete = PatchCorrector.applyPatch(patchList.subList(0, indexCandidateDeletePatch), text)
         val finishText = PatchCorrector.applyPatch(patchList, text)
-        val textNoPatch = PatchCorrector.applyPatch(
+        textNoPatch = PatchCorrector.applyPatch(
                 PatchCorrector.deletePatch(
                         patchList[indexCandidateDeletePatch],
                         patchList.subList(indexCandidateDeletePatch + 1, patchList.size),
-                textBeforeCandidateDelete),
+                        textBeforeCandidateDelete),
                 finishText
-        )
-        val response = CosmasProto.getTextWithoutPatchResponse.newBuilder()
+            )
+        } catch (e : PatchCorrector.ApplyPatchException) {
+            val status = Status.ABORTED.withDescription(e.message)
```

Comment:
ABORTED has very specific meaning and I suggest using just INTERNAL.
Besides, you really want to log the exception here to error log. Otherwise you will have no even minimal debug information in production.

Reply:
done